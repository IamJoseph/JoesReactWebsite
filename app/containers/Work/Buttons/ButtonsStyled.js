import styled from 'styled-components';
import { Link as Linked } from 'react-router-dom';

export const Container = styled.div`
  flex: 1.5;
  flex-wrap: wrap;
  font-weight: bold;
  height: 85%;
  list-style-type: none;
  margin-top: 85px;
  text-align: center;
  @media(min-width: 768px){
    flex: 1;
  }
`;

const buttonColor = (type, hue, active) => {
  switch (type) {
    case 'borderTop':
      if (active) {
        return `hsl(${hue}, 50%, 72%)`;
      }
      return `hsl(${hue}, 69%, 82%)`;
    case 'borderBottom':
      if (active) {
        return `hsl(${hue}, 38%, 69%)`;
      }
      return `hsl(${hue}, 55%, 76%)`;
    case 'boxShadow':
      return `hsl(${hue}, 41%, 73%)`;
    case 'background':
    default:
      return `hsl(${hue}, 70%, 78%)`;
  }
};

export const LI = styled.li`
  background-color: ${({ color }) => buttonColor('background', color)};
  border-bottom: ${({ color, active }) => `3px solid ${buttonColor('borderBottom', color, active)}`};
  border-bottom-left-radius: ${({ bottomRadius }) => bottomRadius ? '5px' : 0};
  border-top: ${({ color, active }) => `3px solid ${buttonColor('borderTop', color, active)}`};
  border-top-left-radius: ${({ topRadius }) => topRadius ? '5px' : 0};
  box-shadow: ${({ color, active, boxShadow }) => {
    if (active && boxShadow) {
      return `0px -1px 20px 0px ${buttonColor('boxShadow', color)} inset, 2px 4px 5px -2px #bacaaa`;
    } else if (boxShadow) {
      return '2px 4px 5px -2px #bacaaa';
    } else if (active) {
      return `0px -1px 20px 0px ${buttonColor('boxShadow', color)} inset`;
    }
    return '';
  }};
  cursor: pointer;
  padding: 0;
`;

export const Link = styled(Linked)`
  color: #fffef6;
  display: inline-block;
  height: 100%;
  padding: 1em 0;
  text-decoration: none;
  user-select: none;
  width: 100%;
`;
