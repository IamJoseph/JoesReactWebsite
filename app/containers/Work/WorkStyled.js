import styled from 'styled-components';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';

export const Container = styled.section`
background-color: #e4e2d7;
border: 1.5px solid #c2cdd6;
display: flex;
height: 100%;
justify-content: center;
margin-left: 25px;
margin-right: 25px;
max-width: 1460px;
padding: 0 .5em;
width: 100%;
z-index: 1;
@media(min-width: 768px){
  padding: 0 1em;
}
@media(min-width: 1150px){
  padding: 0 1.87em;
}
`;

export const PageContainer = styled(CSSTransitionGroup)`
  background: linear-gradient(to bottom, #f9f9f1 50%,#edebe1 90%, #edebe1 100%);
  border: 1.5px solid #9fb4c6;
  border-radius: 0 5px 5px 5px;
  box-shadow: 2px 2px 5px #bdbcb5;
  display: flex;
  flex: 8;
  height: 85%;
  justify-content: center;
  margin-top: 85px;
  max-height: calc(100% - 100px);
  overflow-y: auto;
  padding: 1em;
`;
