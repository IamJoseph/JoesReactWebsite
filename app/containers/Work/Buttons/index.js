import React from 'react';
import PropTypes from 'prop-types';
import { Container, LI, Link } from './ButtonsStyled';

export default function Buttons({ active }) {
  return (
    <Container>
      <LI color="198" active={active === '/work'} topRadius>
        <Link to="/work">Jobs</Link>
      </LI>
      <LI color="183" active={active === '/work/projects'}>
        <Link to="/work/projects">Projects</Link>
      </LI>
      <LI color="155" active={active === '/work/current_site'}>
        <Link to="/work/current_site">Current Site</Link>
      </LI>
      <LI color="97" active={active === '/work/client_sites'}>
        <Link to="/work/client_sites">Client Sites</Link>
      </LI>
      <LI color="65" active={active === '/work/graphics'} bottomRadius boxShadow>
        <Link to="/work/graphics">Graphics</Link>
      </LI>
    </Container>
  );
}

Buttons.propTypes = {
  active: PropTypes.string.isRequired,
};
