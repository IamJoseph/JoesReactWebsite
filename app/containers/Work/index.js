/*
 *
 * Work
 *
 */

import React, { PureComponent } from 'react';
import Helmet from 'react-helmet';
import PropTypes from 'prop-types';
import { Switch } from 'react-router-dom';
import { Container, PageContainer } from './WorkStyled';
import Buttons from './Buttons';
import AsyncRoute from '../../routing/AsyncRoute';
import currentLoader from './Pages/Current/loader';
import clientLoader from './Pages/Client/loader';
import graphicsLoader from './Pages/Graphics/loader';
import defaultLoader from './Pages/Default/loader';
import projectsLoader from './Pages/Projects/loader';

export default class Work extends PureComponent { // eslint-disable-line react/prefer-stateless-function
  getSubRouteContent = (path) => {
    const { store } = this.context;
    switch (path) {
      case '/work/current_site':
        return currentLoader(store);
      case '/work/client_sites':
        return clientLoader(store);
      case '/work/projects':
        return projectsLoader(store);
      case '/work/graphics':
        return graphicsLoader(store);
      default:
      case '/work':
        return defaultLoader(store);
    }
  }

  render() {
    const { pathname } = this.props.location;
    return (
      <Container>
        <Helmet>
          <title>Work</title>
          <meta name="description" content="Past employment, projects, skills" />
        </Helmet>
        <Buttons active={pathname} />
        <PageContainer
          component="div"
          className="transition-group"
          transitionName="page"
          transitionEnterTimeout={2000}
          transitionLeave={false}
          transitionAppear
          transitionAppearTimeout={2000}
        >
          {/* transition needs rendered element on pageload */}
          <div style={{ width: '100%', height: '100%' }}>
            <Switch>
              <AsyncRoute path={pathname} load={this.getSubRouteContent(pathname)} />
            </Switch>
          </div>
        </PageContainer>
      </Container>
    );
  }
}

Work.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }),
};

Work.defaultProps = {
  location: {
    pathname: '',
  },
};

Work.contextTypes = {
  store: PropTypes.object,
};
