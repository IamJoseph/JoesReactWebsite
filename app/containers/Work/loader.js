/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work')
    .then(cb)
    .catch(errorLoading);
};
