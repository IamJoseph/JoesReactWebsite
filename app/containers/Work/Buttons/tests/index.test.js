import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MemoryRouter from 'react-router/MemoryRouter';
import Buttons from '../index';
import Work from '../../index';

describe('<Buttons />', () => {
  it('should render correctly', () => {
    const fakeProps = {
      active: '/work',
    };
    const wrapper = shallow(
      <Buttons {...fakeProps} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render work button correctly', () => {
    const fakeProps = {
      location: {
        pathname: '/work',
      },
    };
    const wrapper = mount(
      <MemoryRouter>
        <Work {...fakeProps} />
      </MemoryRouter>
    );
    const button = wrapper.find('li').first();
    expect(toJson(button)).toMatchSnapshot();
  });

  it('should render graphics button correctly', () => {
    const fakeProps = {
      location: {
        pathname: '/work/graphics',
      },
    };
    const wrapper = mount(
      <MemoryRouter>
        <Work {...fakeProps} />
      </MemoryRouter>
    );
    const button = wrapper.find('li').last();
    expect(toJson(button)).toMatchSnapshot();
  });
});
