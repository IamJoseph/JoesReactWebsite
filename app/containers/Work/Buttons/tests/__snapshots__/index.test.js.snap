// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> should render correctly 1`] = `
<styled.div>
  <styled.li
    active={true}
    color="198"
    topRadius={true}
  >
    <Styled(Link)
      to="/work"
    >
      Jobs
    </Styled(Link)>
  </styled.li>
  <styled.li
    active={false}
    color="183"
  >
    <Styled(Link)
      to="/work/projects"
    >
      Projects
    </Styled(Link)>
  </styled.li>
  <styled.li
    active={false}
    color="155"
  >
    <Styled(Link)
      to="/work/current_site"
    >
      Current Site
    </Styled(Link)>
  </styled.li>
  <styled.li
    active={false}
    color="97"
  >
    <Styled(Link)
      to="/work/client_sites"
    >
      Client Sites
    </Styled(Link)>
  </styled.li>
  <styled.li
    active={false}
    bottomRadius={true}
    boxShadow={true}
    color="65"
  >
    <Styled(Link)
      to="/work/graphics"
    >
      Graphics
    </Styled(Link)>
  </styled.li>
</styled.div>
`;

exports[`<Buttons /> should render graphics button correctly 1`] = `
<li
  className="kSwrIg"
  color="65"
>
  <Styled(Link)
    to="/work/graphics"
  >
    <Link
      className="ipJOlB"
      replace={false}
      to="/work/graphics"
    >
      <a
        className="ipJOlB"
        href="/work/graphics"
        onClick={[Function]}
      >
        Graphics
      </a>
    </Link>
  </Styled(Link)>
</li>
`;

exports[`<Buttons /> should render work button correctly 1`] = `
<li
  className="kmoYbu"
  color="198"
>
  <Styled(Link)
    to="/work"
  >
    <Link
      className="ipJOlB"
      replace={false}
      to="/work"
    >
      <a
        className="ipJOlB"
        href="/work"
        onClick={[Function]}
      >
        Jobs
      </a>
    </Link>
  </Styled(Link)>
</li>
`;
